### ISSUE TITLE
Error: cannot determine type of empty array

### ISSUE THREAD

**Issue by cbandy** [postgraphql/postgraphql#406]
When I call the following mutation with an empty array, I get an error from postgres about an untyped array.

```sql
CREATE TABLE graphql.role (
  role_name  text PRIMARY KEY,
  operations text[] NOT NULL
);
```

```graphql
mutation CreateRole($operations: [String!]!) {
  createRole(input: {role: {roleName: "chris", operations: $operations}}) {
    role {
      roleName
      operations
    }
  }
}
```

```
STATEMENT:  with __local_0__ as ( insert into "graphql"."role" ("role_name", "operations") values ($1, array[]) returning * ) select row_to_json(__local_0__) as object from __local_0__
ERROR:  cannot determine type of empty array at character 92
HINT:  Explicitly cast to the desired type, for example ARRAY[]::integer[].
```

Using an array literal works well in `psql`:

```
psql> with __local_0__ as ( insert into "graphql"."role" ("role_name", "operations") values ('chris', '{}') returning * ) select row_to_json(__local_0__) as object from __local_0__;
                object                 
---------------------------------------
 {"role_name":"chris","operations":[]}
(1 row)
```

When I call the mutation with a non-empty array, I can see the generated query is

```sql
with __local_0__ as ( insert into "graphql"."role" ("role_name", "operations") values ($1, array[$2, $3]) returning * ) select row_to_json(__local_0__) as object from __local_0__
```

Is it possible for `postgraphql` to use `pg` to serialize arrays?

(I'm willing, though perhaps not able, to contribute a fix. Where should I look for the generation of this query?)

**Comment by benjie** [comment]
May be better to fall back to literal `'{}'` instead of `array[]` when empty?

**Comment by cbandy** [comment]
@benjie good idea. I've opened a PR using this approach.

**Comment by calebmer** [comment]
Fixed in 3.1.0 🎉
